src/index.ts

```diff
@@ -1,54 +1,95 @@
-import { fetchVoteData } from "./queries/voting-data";
-const Web3 = require("web3");
-require("dotenv").config();
-const fs = require("fs");
+import { fetchVoteData } from './queries/voting-data';
+import { BigNumber, ethers } from 'ethers';
+import { expect } from 'chai';
+import { describe } from 'node:test';
+require('dotenv').config();
+const Web3 = require('web3');
+const fs = require('fs');
+const zero = ethers.constants.Zero;
 
-if(!process.env.VOTE_ID) {
-  throw Error("Vote ID is required as an env variable")
+if (!process.env.VOTE_ID) {
+  throw Error('Vote ID is required as an env variable');
 }
+// id of the vote to query
+const voteId: string = process.env.VOTE_ID;
+// TODO: This will come from the collection contract
+const DISTRIBUTION_AMOUNT = ethers.utils.parseEther('100');
+// helper variable used to achieve a good decimal approximation in the rewards distribution calculation
+const factor = DISTRIBUTION_AMOUNT.div(100000);
 
-// This is the vote we are distributing rewards for
-const voteId = process.env.VOTE_ID
-
-// TODO: This will come from the collections contract
-const DISTRIBUTION_AMOUNT = 10000
-
+// fetch SARCO staking contract that gives stakers voting rights tokens (SarcoVR)
 const web3 = new Web3(
   new Web3.providers.HttpProvider(
     `https://${process.env.ETHEREUM_NETWORK}.infura.io/v3/${process.env.INFURA_API_KEY}`
   )
 );
+const stakingContractABI = JSON.parse(fs.readFileSync('src/abi/sarcoStaking.json'));
+const stakingContract = new web3.eth.Contract(stakingContractABI, process.env.CONTRACT_ADDRESS);
 
-const stakingContractABI  = JSON.parse(fs.readFileSync("src/abi/sarcoStaking.json"));
-const stakingContract = new web3.eth.Contract(
-  stakingContractABI,
-  process.env.CONTRACT_ADDRESS
-);
+// helper function that sums the BN values of a mapping: used to check the sum of rewards equals the initial amount distributed
+function getSum(distribution: Map<string, BigNumber>): BigNumber {
+  let sum = zero;
+  for (let i of distribution.keys()) {
+    let value: BigNumber = distribution.get(i);
+    sum = sum.add(value);
+  }
+  return sum;
+}
 
```

Comment:
Since `voteData` is being defined on line 68 under the `main` function, it should be passed in as an argument of this function so that `fetchVoteData` is only called once.